cmd/cluster/validatepullsecret.go

```diff
@@ -0,0 +1,92 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var BackplaneClusterAdmin = "backplane-cluster-admin"
+
+func newCmdPullSecret(kubeCli client.Client, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	return &cobra.Command{
+		Use:               "validate-pull-secret",
+		Short:             "Checks if the pull secret email matches the owner email",
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ValidatePullSecret(args[0], kubeCli, flags))
+		},
+	}
+}
+
+func ValidatePullSecret(clusterID string, kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
+	ocm := utils.CreateConnection()
+	defer func() {
+		if ocmCloseErr := ocm.Close(); ocmCloseErr != nil {
+			fmt.Printf("Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
+		}
+	}()
+
+	fmt.Println("Checking if pull secret email matches user email")
+
+	// This is the flagset for the kubeCli object provided from the root command. Set here to retroactively impersonate backplane-cluster-admin
+	flags.Impersonate = &BackplaneClusterAdmin
+	secret := &corev1.Secret{}
+	err := kubeCli.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-config", Name: "pull-secret"}, secret)
+	if err != nil {
+		return err
+	}
+
+	type auth struct {
+		Email string `json:"email"`
+	}
+
+	type auths struct {
+		Cloudopenshiftcom auth `json:"cloud.openshift.com"`
+	}
+
+	type response struct {
+		Auths auths `json:"auths"`
+	}
+
+	dockerConfigJson := response{}
+	err = json.Unmarshal(secret.Data[".dockerconfigjson"], &dockerConfigJson)
+	if err != nil {
+		return err
+	}
+
+	clusterPullSecretEmail := dockerConfigJson.Auths.Cloudopenshiftcom.Email
+	if clusterPullSecretEmail == "" {
+		fmt.Printf("%v\n%v\n%v\n",
+			"Couldn't extract email address from pull secret for cloud.openshift.com",
+			"This can mean the pull secret is misconfigured. Please verify the pull secret manually:",
+			"  oc get secret -n openshift-config pull-secret -o json | jq -r '.data[\".dockerconfigjson\"]' | base64 -d")
```

Comment:
Maybe we could handle this manual validation with `osdctl`?  We have the cluster id, `osdctl` can login to clusters.  Could we just have it do so here?

Not a blocker for me - maybe worth looking at in the future.

Reply:
Could you explain a bit more about what you'd like to see here? What we're doing above in code is the equivalent of what's written here in this output, followed up by trying to find the email address in `.dockerconfigjson` which fails. At that point I'm not sure how else we could programmatically recover.

Reply:
Just ignore me today, I guess.  :facepalm:  Apparently my brain is not here at all.